src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    const len = this.length;
+    let joinString = '';
+
+    if (len !== 0) {
```

Comment:
```suggestion
    if (len !== 0) {
```
This check is redundant. If `this.length === 0` then loop will not even start:
'for (let i = 0; i < 0; i++)'
0 will never be less than 0 so the loop will not have even one iteration